component-samples/demo/scripts/fdo_toolkit.sh

```diff
@@ -0,0 +1,282 @@
+        echo "Found java 11"
+else
+        echo "Java 11 not found, installing it..."
+        apt-get update && apt-get install -y openjdk-11-jre-headless
+        chk $? 'installing java 11'
+fi
+
+
+if ! dpkg-query -s haveged > /dev/null 2>&1; then
+    echo "Haveged is required, installing it"
+    sudo apt-get install -y haveged
+    chk $? 'installing haveged'
+fi
+
+# If docker isn't installed, do that
+if ! command -v docker >/dev/null 2>&1; then
+    echo "Docker is required, installing it..."
+    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+    chk $? 'adding docker repository key'
+    add-apt-repository "deb [arch=$(dpkg --print-architecture)] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+    chk $? 'adding docker repository'
+    apt-get install -y docker-ce docker-ce-cli containerd.io
+    chk $? 'installing docker'
+fi
+
+
+# If docker-compose isn't installed, or isn't at least 1.21.0 (when docker-compose.yml version 2.4 was introduced), then install/upgrade it
+# For the dependency on 1.21.0 or greater, see: https://docs.docker.com/compose/release-notes/
+minVersion=1.29.2
+if ! isDockerComposeAtLeast $minVersion; then
+    if [[ -f '/usr/bin/docker-compose' ]]; then
+        echo "Error: Need at least docker-compose $minVersion. A down-level version is currently installed, preventing us from installing the latest version. Uninstall docker-compose and rerun this script."
+        exit 2
+    fi
+    echo "docker-compose is not installed or not at least version $minVersion, installing/upgrading it..."
+    # Install docker-compose from its github repo, because that is the only way to get a recent enough version
+    curl --progress-bar -L "https://github.com/docker/compose/releases/latest/download/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+    chk $? 'downloading docker-compose'
+    chmod +x /usr/local/bin/docker-compose
+    chk $? 'making docker-compose executable'
+    ln -s /usr/local/bin/docker-compose /usr/bin/docker-compose
+    chk $? 'linking docker-compose to /usr/bin'
+fi
+
+
+echo "Upgraded and Installed all FDO dependencies"
+}
+
+update_scripts() {
+cd $1
+sed -i 's/#subjectAltName = @alt_names/subjectAltName = @alt_names/g' web-server.conf
+sed -i 's/#subjectAltName/subjectAltName/g' web-server.conf
+head -n -8 web-server.conf > temp.txt ; mv temp.txt web-server.conf
+
+echo "[ alt_names ]">> web-server.conf
+echo "DNS.1 = $ipaddress" >> web-server.conf
+echo "DNS.2 = localhost" >> web-server.conf
+echo "DNS.3 = host.docker.internal" >> web-server.conf
+echo "DNS.4 = host.docker.internal" >> web-server.conf
+echo "IP.1 = $ipaddress" >> web-server.conf
+echo "IP.2 = 127.0.0.1" >> web-server.conf
+echo "IP.3 = 172.17.0.1" >> web-server.conf
+
+}
+
+generate_keys() {
+cd $1
+bash demo_ca.sh
+bash web_csr_req.sh
+bash user_csr_req.sh
+bash keys_gen.sh
+chmod -R 777 secrets/ service.env
```

Comment:
To avoid all (r,w,x) permissions to all files, grant only read permission for server-key.pem

```suggestion
chmod +r secrets/server-key.pem
```